dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/ApiExtractor.java

```diff
@@ -553,14 +521,17 @@ private void extractParams(Api.Endpoint endpoint, Class<?> paramsObject) {
     String sharedName = getSharedName(paramsObject, shared, null);
     if (sharedName != null) {
       Map<Class<?>, List<Api.Parameter>> sharedParameters = api.getComponents().getParameters();
+      boolean addShared = !sharedParameters.containsKey(paramsObject);
```

Comment:
FYI: without this check the same params class would add all its parameters each time the params class occurred in an endpoint method resulting in 18K parameters instead of ~250. Woopsy! :sweat_smile: